business_objects/upload_task.py

```diff
@@ -49,18 +55,39 @@ def create(
         file_type=file_type,
         file_import_options=file_import_options,
         upload_type=upload_type,
+        key=key,
     )
     general.add(task, with_commit)
     return task
 
 
+def remove_key(
+    project_id: str,
+    task_id: str,
+    with_commit: bool = False,
+) -> None:
+    task: UploadTask = get(project_id, task_id)
+    task.key = None
+    general.flush_or_commit(with_commit)
+
+
+def remove_all_keys(
+    with_commit: bool = False,
+) -> None:
+    tasks = get_all_global()
+    for task in tasks:
+        task.key = None
+    general.flush_or_commit(with_commit)
+
+
 def update(
     project_id: str,
     task_id: str,
     state: Optional[str] = None,
     progress: Optional[float] = None,
     file_additional_info: str = None,
     mappings: Optional[Dict[str, Any]] = None,
+    key: Optional[str] = None,
```

Comment:
not sure this will work as expected if we want to update a task from key to None. 
Maybe we need something like a "NO_VALUE" as default and if it's != "NO_VALUE" we set it (even None) or the other way around and if it's == NO_VALUE we set it to None in the task.key= key part.

either way a comment would be good so the next person looking at this isn't irritated

Reply:
Yeah true, I wrote a dedicated method for removing keys but forgot to remove the parameter from this signature